src/App.js

```diff
@@ -1,23 +1,127 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+
+import babyNamesData from "./babyNamesData.json";
+import BabyNamesCard from "./BabyNamesCard";
+import SearchBar from "./SearchBar";
+import FavoriteNames from "./FavoriteNames";
+import ResetButton from "./ResetButton";
+import { useEffect, useState } from "react";
+import GenderSelector from "./GenderSelector";
 
 function App() {
+  let [favoriteName, setFavoriteName] = useState([]);
+  let [noneFavoriteNames, setNoneFavoriteNames] = useState(babyNamesData);
+  let [isClickingFavoriteNames] = useState(true);
+  const [showFavoriteNameMessage, setShowFavoriteNameMessage] = useState(true);
+  let storedFavoriteNames = localStorage.getItem("storedFavoriteNames");
+  let storedNoneFavoriteNames = localStorage.getItem("storedNoneFavoriteNames");
+
+  useEffect(() => {
+    if (storedFavoriteNames && storedNoneFavoriteNames) {
+      setFavoriteName(JSON.parse(storedFavoriteNames));
+      setNoneFavoriteNames(JSON.parse(storedNoneFavoriteNames));
+    }
+  }, [storedFavoriteNames, storedNoneFavoriteNames]);
+  function filterNames(e) {
+    let searchInputValue = e.target.value.toUpperCase();
+    let filteredBabyNames = babyNamesData.filter((baby) => {
+      return baby.name.toUpperCase().includes(searchInputValue);
+    });
+    setNoneFavoriteNames(filteredBabyNames);
+  }
+  useEffect(() => {
+    let favoriteNamesMessageShow = favoriteName.length > 0;
+    console.log(favoriteNamesMessageShow)
+    setShowFavoriteNameMessage(favoriteNamesMessageShow);
+  }, [favoriteName]);
+  function moveToFavorite(e) {
+    setFavoriteName();
+    let name = e.target.value;
+    let chosenName = babyNamesData.filter((baby) => {
+      return baby.name === name;
+    });
+    chosenName = [...favoriteName, ...chosenName];
+    noneFavoriteNames = noneFavoriteNames.filter((baby) => {
+      return baby.name !== name;
+    });
+    setNoneFavoriteNames(noneFavoriteNames);
+    setFavoriteName(chosenName);
+    localStorage.setItem("storedFavoriteNames", JSON.stringify(chosenName));
+    localStorage.setItem(
+      "storedNoneFavoriteNames",
+      JSON.stringify(noneFavoriteNames)
+    );
+  }
+  function moveFromFavorite(e) {
+    let name = e.target.value;
+    let favoriteNames = favoriteName.filter((baby) => {
+      return baby.name !== name;
+    });
+    let addNonefavorite = favoriteName.filter((baby) => {
+      return baby.name === name;
+    });
+    noneFavoriteNames = [...noneFavoriteNames, ...addNonefavorite];
+    setNoneFavoriteNames(noneFavoriteNames);
+    setFavoriteName(favoriteNames);
+    localStorage.setItem("storedFavoriteNames", JSON.stringify(favoriteNames));
+    localStorage.setItem(
+      "storedNoneFavoriteNames",
+      JSON.stringify(noneFavoriteNames)
+    );
+  }
+  const [genderPictureClass, setGenderPictureClass] = useState([
```

Comment:
It's quite difficult to understand the purpose of an array of booleans. It would be clearer to make this an object, i.e.

```js
{
    bothgenders: false,
    boy: false,
    girl: false
}
```

Reply:
I couldn't think that. thank you very much absolutely more meaningful